aws/data_source_aws_ec2_instance_type_test.go

```diff
@@ -0,0 +1,179 @@
+					resource.TestCheckResourceAttr(resourceMetal, "total_instance_storage", "60000"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsEc2InstanceType_gpu(t *testing.T) {
+	resourceGpu := "data.aws_ec2_instance_type.gpu"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceEc2InstanceTypeGpu,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceGpu, "gpus.#", "1"),
+					resource.TestCheckResourceAttr(resourceGpu, "gpus.0.count", "1"),
+					resource.TestCheckResourceAttr(resourceGpu, "gpus.0.manufacturer", "NVIDIA"),
+					resource.TestCheckResourceAttr(resourceGpu, "gpus.0.memory_size", "4096"),
+					resource.TestCheckResourceAttr(resourceGpu, "gpus.0.name", "K520"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsEc2InstanceType_fpga(t *testing.T) {
+	resourceFpga := "data.aws_ec2_instance_type.fpga"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceEc2InstanceTypeFgpa,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceFpga, "fpgas.#", "1"),
+					resource.TestCheckResourceAttr(resourceFpga, "fpgas.0.count", "1"),
+					resource.TestCheckResourceAttr(resourceFpga, "fpgas.0.manufacturer", "Xilinx"),
+					resource.TestCheckResourceAttr(resourceFpga, "fpgas.0.memory_size", "65536"),
+					resource.TestCheckResourceAttr(resourceFpga, "fpgas.0.name", "Virtex UltraScale (VU9P)"),
+					resource.TestCheckResourceAttr(resourceFpga, "total_fpga_memory", "65536"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsEc2InstanceType_accelerator(t *testing.T) {
+	resourceAccelerator := "data.aws_ec2_instance_type.accelerator"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceEc2InstanceTypeAccelerator,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceAccelerator, "inference_accelerators.#", "1"),
+					resource.TestCheckResourceAttr(resourceAccelerator, "inference_accelerators.0.count", "1"),
+					resource.TestCheckResourceAttr(resourceAccelerator, "inference_accelerators.0.manufacturer", "AWS"),
+					resource.TestCheckResourceAttr(resourceAccelerator, "inference_accelerators.0.name", "Inferentia"),
+				),
+			},
+		},
+	})
+}
+
+const testAccDataSourceEc2InstanceTypeBasic = `
+data "aws_ec2_instance_type" "basic" {
+  instance_type = "t2.micro"
+}
+`
+
+const testAccDataSourceEc2InstanceTypeMetal = `
+data "aws_ec2_instance_type" "metal" {
+  instance_type = "i3en.metal"
+}
+`
+
+const testAccDataSourceEc2InstanceTypeGpu = `
+data "aws_ec2_instance_type" "gpu" {
+  instance_type = "g2.2xlarge"
```

Comment:
```suggestion
  instance_type = "g3.4xlarge"
```
My apologies but I just noticed that `g2` is not available in some partitions.